sdk/storage/azure-storage-file-datalake/azure/storage/filedatalake/_download.py

```diff
@@ -3,26 +3,34 @@
 # Licensed under the MIT License. See License.txt in the project root for
 # license information.
 # --------------------------------------------------------------------------
-from typing import IO, Iterator, Optional
+from typing import (
+    Any, cast, Generic, IO, Iterator, TypeVar,
+    TYPE_CHECKING
+)
 
 from ._deserialize import from_blob_properties
 
+if TYPE_CHECKING:
+    from ._models import FileProperties
 
-class StorageStreamDownloader(object):
-    """A streaming object to download from Azure Storage.
 
-    :ivar str name:
-        The name of the file being downloaded.
-    :ivar ~azure.storage.filedatalake.FileProperties properties:
-        The properties of the file being downloaded. If only a range of the data is being
-        downloaded, this will be reflected in the properties.
-    :ivar int size:
-        The size of the total data in the stream. This will be the byte range if specified,
-        otherwise the total size of the file.
-    """
+T = TypeVar('T', bytes, str)
 
-    def __init__(self, downloader):
-        self._downloader = downloader
+
+class StorageStreamDownloader(Generic[T]):
```

Comment:
This download should not be generic like Blob. Everything in Datalake just returns `bytes`